### DISCUSSION TITLE
Nextcloud container slow start

### DISCUSSION THREAD

**Discussion by attssystem** [nextcloud/all-in-one#2076] [Questions] [answered]
Hi,

I need help regarding slow starts of my nextcloud container. Mastercontainer starts fast and others too but Nextcloud one struggles for approx. half an hour at each start. I know the question had been asked (https://github.com/nextcloud/all-in-one/discussions/1600) but it's marked as answered and I couldn't find a solution in my case. I tried running struggling commands (from entrypoint.sh) directly in the container without any problem. There's no error, it's just very slow at start then everything is ok. Does someone have an idea ? :)

Hugues

**Comment by szaimen** [comment]
This was the solution for them: https://github.com/nextcloud/all-in-one/discussions/1600#discussioncomment-4596033

**Reply by attssystem** [reply]
I saw that but it didn't solve my issue . In fact, I am in a similar situation. I have my main router and Nextcloud runs behind a second router in my DMZ. In my DMZ, Nextcloud is only reachable through an apache proxy. But I have a basic firewall, it forwards traffic to my proxy that reaches my Nextcloud instance. Nextcloud container can reach internet without any issue. I do not know where to look at as logs do not help in that case.

**Reply by szaimen** [reply]
Sorry, no idea what the problem could be then. I still think it is your network infrastructure that leads to this

**Reply by attssystem** [reply]
I understand. Don't you think there's at least a place I should look at for more logging ?

**Comment by szaimen** [comment]
Can you check if you are effected by https://github.com/nextcloud/all-in-one/discussions/2065

**Reply by attssystem** [reply]
My nextcloud instance is only a few weeks ago so I shouldn't be concerned. Nevertheless, I applied the provided correction as it couldn't broke my installation and it didn't solve my issue. By the way, ```docker network inspect nextcloud-aio | grep Internal``` returns ```"Internal": false,``` so it doesn't seem to be the issue. Thanks for the idea.

Below are my logs if it can help. All containers are enabled and this time, the container took 25 minutes.

```
              now              
-------------------------------
 2023-02-27 20:16:49.593625+00
(1 row)

+ '[' -f /dev-dri-group-was-added ']'
++ find /dev -maxdepth 1 -mindepth 1 -name dri
+ '[' -n '' ']'
+ set +x
Installing imagemagick via apk...
Enabling Imagick...
Configuring Redis as session handler...
Setting php max children...
System config value tempdirectory set to string /mnt/ncdata/tmp/
Applying one-click-instance settings...
System config value one-click-instance set to boolean true
System config value one-click-instance.user-limit set to integer 100
System config value one-click-instance.link set to string https://nextcloud.com/all-in-one/
support already enabled
Adjusting log files...
System config value logfile set to string /var/www/html/data/nextcloud.log
Config value logfile for app admin_audit set to /var/www/html/data/audit.log
Applying network settings...
System config value trusted_domains => 1 set to string cloud.xxxxx.xx
System config value overwrite.cli.url set to string https://cloud.xxxxx.xx/
System config value htaccess.RewriteBase set to string /
.htaccess has been updated
System config value files_external_allow_create_new_local set to boolean false
System config value trusted_proxies => 0 set to string 127.0.0.1
System config value trusted_proxies => 1 set to string ::1
Config value base_endpoint for app notify_push set to https://cloud.xxxxx.xx/push
Config value wopi_url for app richdocuments set to https://cloud.xxxxx.xx/
System config value allow_local_remote_servers set to boolean true
No ipv6-address found for cloud.xxxxxxxxx.
Config value wopi_allowlist for app richdocuments set to 2400:cb00::/32,2606:4700::/32,2803:f800::/32,2405:b500::/32,2405:8100::/32,2a06:98c0::/29,2c0f:f248::/32,173.245.48.0/20,103.21.244.0/22,103.22.200.0/22,103.31.4.0/22,141.101.64.0/18,108.162.192.0/18,190.93.240.0/20,188.114.96.0/20,197.234.240.0/22,198.41.128.0/17,162.158.0.0/15,104.16.0.0/13,104.24.0.0/14,172.64.0.0/13,131.0.72.0/22,188.114.97.2,127.0.0.1/8,192.168.0.0/16,172.16.0.0/12,10.0.0.0/8,fd00::/8,::1,188.114.96.2
Config value av_mode for app files_antivirus set to daemon
Config value av_port for app files_antivirus set to 3310
Config value av_host for app files_antivirus set to nextcloud-aio-clamav
Config value av_stream_max_length for app files_antivirus set to 104857600
Config value av_max_file_size for app files_antivirus set to -1
Config value av_infected_action for app files_antivirus set to only_log
System config value enabledPreviewProviders => 0 set to string OC\Preview\Imaginary
System config value preview_imaginary_url set to string http://nextcloud-aio-imaginary:9000
{
    "search_platform": "OCA\\FullTextSearch_Elasticsearch\\Platform\\ElasticSearchPlatform",
    "app_navigation": "0",
    "provider_indexed": "",
    "cron_err_reset": "1677054347",
    "tick_ttl": "1800",
    "collection_indexing_list": "50",
    "migration_24": "1"
}
{
    "elastic_host": "http:\/\/nextcloud-aio-fulltextsearch:9200",
    "elastic_index": "nextcloud-aio",
    "fields_limit": "10000",
    "es_ver_below66": "0",
    "analyzer_tokenizer": "standard"
}
{
    "files_local": "1",
    "files_external": "0",
    "files_group_folders": "0",
    "files_encrypted": "0",
    "files_federated": "0",
    "files_size": "20",
    "files_pdf": "1",
    "files_office": "1",
    "files_image": "0",
    "files_audio": "0"
}
[27-Feb-2023 20:41:16] NOTICE: fpm is running, pid 881
[27-Feb-2023 20:41:16] NOTICE: ready to handle connections
```

Thanks for your help ;)

**Reply by szaimen** [reply]
> My nextcloud instance is only a few weeks ago so I shouldn't be concerned.

Yes indeed.

**Reply by szaimen** [reply]
Can you try to adjust the MTU of your Router? At least for Wireguard this potentially fixes the problem.

**Comment by ladislavkrivy** [comment]
I may be in the same boat and need help. 

My nextcloud is inaccessible. Aio-apache  container was reported unhealthy. In the logs there was the warning for caddy buffer size so I fixed that and went to sleep. Now the next day it's still inaccessible (aio-apache unhealthy) and in nextcloud logs I'm getting:
```
ERROR [notify_push] src/main.rs:77: Self test failed: Error while communicating with nextcloud instance
```

Investigating further, after rebooting everything I noticed the slow start as well. The nextcloud log is rolling out really slowly so I suppose there is some connection timing out. My network setup is even weirder - I'm behind cg nat so I've set up wireguard tunnel to a vps. On the vps I forward port 443 traffic the home server and on the home server i push all reply traffic trough the vps. (mark the incoming 443 packets and route the outgoing marked packets through vps. That way the vps sees just encrypted trffic)

Maybe if there is a way to turn off the connectivity check that may be causing our issue that would help both of us. Is there somewhere a log with the timeouts? Anny suggestion what should I investigate next? 

Complete logs below:
nextcloud:
```
              now              
-------------------------------
 2023-02-28 08:38:53.737806+01
(1 row)

+ '[' -f /dev-dri-group-was-added ']'
++ find /dev -maxdepth 1 -mindepth 1 -name dri
+ '[' -n '' ']'
+ set +x
Installing imagemagick via apk...
Enabling Imagick...
Configuring Redis as session handler...
Setting php max children...
System config value tempdirectory set to string /mnt/ncdata/tmp/
Applying one-click-instance settings...
System config value one-click-instance set to boolean true
System config value one-click-instance.user-limit set to integer 100
System config value one-click-instance.link set to string https://nextcloud.com/all-in-one/
support already enabled
Adjusting log files...
System config value logfile set to string /var/www/html/data/nextcloud.log
Config value logfile for app admin_audit set to /var/www/html/data/audit.log
Applying network settings...
System config value trusted_domains => 1 set to string xxx.tld
System config value overwrite.cli.url set to string https://xxx.tld/
System config value htaccess.RewriteBase set to string /
.htaccess has been updated
System config value files_external_allow_create_new_local set to boolean true
System config value trusted_proxies => 0 set to string 127.0.0.1
System config value trusted_proxies => 1 set to string ::1
Config value base_endpoint for app notify_push set to https://xxx.tld/push
System config value onlyoffice => jwt_secret set to string xxx
System config value onlyoffice => jwt_header set to string xxx
Config value DocumentServerUrl for app onlyoffice set to https://xxx.tld/onlyoffice
System config value allow_local_remote_servers set to boolean true
[28-Feb-2023 08:55:02] NOTICE: fpm is running, pid 567
[28-Feb-2023 08:55:02] NOTICE: ready to handle connections
[2023-02-28 08:57:13.025496 +01:00] ERROR [notify_push] src/main.rs:77: Self test failed: Error while communicating with nextcloud instance
```

apache:
```
Waiting for Nextcloud to start...
...
{"level":"info","ts":1677570910.7240582,"msg":"using provided configuration","config_file":"/Caddyfile","config_adapter":""}
[Tue Feb 28 08:55:11.316405 2023] [mpm_event:notice] [pid 638:tid 140320913025864] AH00489: Apache/2.4.55 (Unix) configured -- resuming normal operations
[Tue Feb 28 08:55:11.316702 2023] [core:notice] [pid 638:tid 140320913025864] AH00094: Command line: '/usr/local/apache2/bin/httpd -D FOREGROUND'
```

**Reply by szaimen** [reply]
Are you using Wireguard for this? I've found this: https://keremerkan.net/posts/wireguard-mtu-fixes/

**Reply by ladislavkrivy** [reply]
Thank you! I tried setting the mtu using `ip link set dev wg_tun mtu 1280` on both sides but unfortunately nextcloud still seems to be starting slowly. I've checked that mtu is set to 1500 on the router but it's possible my ISP did something along the way. I'll play with lower mtu values and try updating the wireguard configs later just to be sure.

Edit: Setting even lower mtu and changing the config and rebooting everything just to be sure did not help.

**Reply by ladislavkrivy** [reply]
I found my issue... It was in my routing setup. Now I feel bad for posting here.

I'm still not sure why it happened. Up until recently, docker was using `docker0` interface for nextcloud and that's how I was marking the packets. (`iptables -t mangle -I PREROUTING -i docker0 -p tcp --sport 443 -j MARK --set-mark 44`) Now a new interface `br-184010c16108` sneaked up and nextcloud traffic is routed to that instead... so the outgoing packets were not properly forwarded trough the vps anymore.

Again sorry and thanks for the advice regarding wireguard MTU size.

**Comment by attssystem** [comment]
For information, last update of containers didn't solved anything. But,,update of the nextcloud container itself only took about 10 minutes, which could mean that it's not affected ? But the start part of it took half an hour.

**Comment by szaimen** [comment, marked as answer]
Can you try to adjust the MTU of your Router? At least for Wireguard this potentially fixes the problem.

**Comment by attssystem** [comment]
Just found the answer. As said in other threads, my container was trying to access my cloud through its public IP. I made a script that modify /etc/hosts at start to solve it.

**Reply by ekuraev** [reply]
Thanks! Add record `127.0.0.1 mydomain ` to `/etc/hosts` fix problem!
